packages/store/src/Slice.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { SchemaType } from "./Types.sol";
+import { Memory } from "./Memory.sol";
+import { DecodeSlice } from "./storagecoder/DecodeSlice.sol";
+
+// Acknowledgements:
+// Based on @dk1a's Slice.sol library (https://github.com/dk1a/solidity-stringutils/blob/main/src/Slice.sol)
+
+// First 16 bytes are the pointer to the data, followed by 16 bytes of data length.
+type Slice is uint256;
+
+using SliceInstance for Slice global;
+using DecodeSlice for Slice global;
+
+/**
+ * @title Static functions for Slice
+ */
+library Slice_ {
```

Comment:
Nit: we moved towards calling these libraries that implement function on a custom type `SliceLib` to make it a bit more explicit what the connection between the library and the type is

Reply:
Renamed; now it's SliceLib for static, SliceInstance for instance funcs (SliceInstance would never need to be imported)

Reply:
I like that - we should probably adjust `SchemaLib` and `PackedCounterLib` to use the same pattern (right now everything is in the `...Lib`)